gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/service/NackAckPreparationService.java

```diff
@@ -0,0 +1,147 @@
+package uk.nhs.adaptors.pss.translator.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.hl7.v3.COPCIN000001UK01Message;
+import org.hl7.v3.RCMRIN030000UK06Message;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.nhs.adaptors.connector.service.MigrationStatusLogService;
+import uk.nhs.adaptors.pss.translator.model.ACKMessageData;
+import uk.nhs.adaptors.pss.translator.model.NACKMessageData;
+import uk.nhs.adaptors.pss.translator.model.NACKReason;
+import uk.nhs.adaptors.pss.translator.task.SendACKMessageHandler;
+import uk.nhs.adaptors.pss.translator.task.SendNACKMessageHandler;
+import uk.nhs.adaptors.pss.translator.util.XmlParseUtil;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class NackAckPreparationService {
+
+    private final SendNACKMessageHandler sendNACKMessageHandler;
+    private final SendACKMessageHandler sendACKMessageHandler;
+    private final MigrationStatusLogService migrationStatusLogService;
+
+    public boolean sendAckMessage(RCMRIN030000UK06Message payload, String conversationId) {
+
+        LOGGER.debug("Sending ACK message for message with Conversation ID: [{}]", conversationId);
+
+        return sendACKMessageHandler.prepareAndSendMessage(prepareAckMessageData(
+                payload,
+                conversationId
+        ));
+    }
+
+    public boolean sendAckMessage(COPCIN000001UK01Message payload, String conversationId, String losingPracticeOdsCode) {
+
+        LOGGER.debug("Sending ACK message for message with Conversation ID: [{}]", conversationId);
+
+        return sendACKMessageHandler.prepareAndSendMessage(prepareAckMessageData(
+                payload,
+                conversationId,
+                losingPracticeOdsCode
+        ));
+    }
+
+    public ACKMessageData prepareAckMessageData(RCMRIN030000UK06Message payload,
```

Comment:
Are those `precapreAckMessageData`  and `prepareNackMessageData` methods called from outside this class? If not then they can be private.